expan/core/experiment.py

```diff
@@ -56,30 +55,30 @@ def analyze_statistical_test(self, test, test_method='fixed_horizon', **worker_a
         if not isinstance(test, StatisticalTest):
             raise TypeError("Statistical test should be of type StatisticalTest.")
 
-        if 'entity' not in self.data.columns:
+        if 'entity' not in test.data.columns:
             raise KeyError("There is no 'entity' column in the data.")
-        if self.data.entity.duplicated().any():
+        if test.data.entity.duplicated().any():
```

Comment:
should we move the duplication check after we perform the feature filter?

Reply:
@shansfolder thanks, will change that!